File: C++/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Exercises/Solutions/P08_OnTimeForTheExam.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <iomanip>
using namespace std;

int main() {

	int examHour, examMinute, arrivalHour, arrivalMinute;
	cin >> examHour >> examMinute >> arrivalHour >> arrivalMinute;

	int exam = examHour * 60 + examMinute;
	int arrival = arrivalHour * 60 + arrivalMinute;

	int diff = abs(exam - arrival);
	if (exam < arrival) {
		cout << "Late" << endl;
		if (60 <= diff) {
			cout << diff / 60 << ":" << setw(2) << setfill('0') << diff % 60 << " hours after the start" << endl;
		}
		cout << diff << " minutes after the start" << endl;

	} else if (arrival <= exam && 0 <= diff && diff <= 30) {
		cout << "On time" << endl;
		if (diff != 0) {
			cout << diff << " minutes before the start" << endl;
		}
	} else if (arrival <= exam && 30 < diff) {
		cout << "Early" << endl;
		if (60 <= diff) {
			cout << diff / 60 << ":" << setw(2) << setfill('0') << diff % 60 << " hours before the start" << endl;
		}
		cout << diff << " minutes before the start" << endl;
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Exercises/Solutions/P08_OnTimeForTheExam.py
hour_exam = int(input())
min_exam = int(input())
hour_arrival = int(input())
min_arrival = int(input())

status = 0

exam_time = (hour_exam * 60) + min_exam
arrival_time = (hour_arrival * 60) + min_arrival

difference = arrival_time - exam_time

if difference > 0:
    status = "Late"
elif difference < -30:
    status = "Early"
elif -30 <= difference <= 0:
    status = "On time"

print(status)

if difference > 0:
    hour_late = difference // 60
    min_late = difference % 60
    if hour_late == 0:
        print(f'{min_late} minutes after the start')
    else:
        print(f'{hour_late}:{min_late:02d} hours after the start')
elif difference == 0:
    print('')
else:
    early = exam_time - arrival_time
    hour_early = early // 60
    min_early = early % 60
    if hour_early == 0:
        print(f'{min_early} minutes before the start')
    else:
        print(f'{hour_early}:{min_early:02d} hours before the start')




